tests/hooks/test_dbt_hook.py

```diff
@@ -1,54 +1,211 @@
+            '--profiles-dir', '/opt/airflow/dags/dbt_project',
+            '--project-dir', '/opt/airflow/dags/project_dir',
+            '--vars', '{"execution_date": "2021-01-01"}', '--select',
+            'my_model']),
+        # test all the params
+        ("dbt", "test", {'profiles_dir': '/path/profiles_folder'},
+            ["dbt", "test", "--profiles-dir", "/path/profiles_folder"]),
+        ("dbt", "run", {'project_dir': '/path/dbt_project_dir'},
+            ["dbt", "run", "--project-dir", "/path/dbt_project_dir"]),
+        ("dbt", "test", {'target': 'model_target'},
+            ["dbt", "test", "--target", "model_target"]),
+        ("dbt", "test", {'vars': {"hello": "world"}},
+            ["dbt", "test", "--vars", '{"hello": "world"}']),
+        ("dbt", "run", {'models': 'my_model'},
+            ["dbt", "run", "--models", "my_model"]),
+        ("dbt", "run", {'exclude': 'my_model'},
+            ["dbt", "run", "--exclude", "my_model"]),
+        ("dbt", "run", {'exclude': 'my_model'},
+            ["dbt", "run", "--exclude", "my_model"]),
+
+        # run specific params
+        ("dbt", "run", {'full_refresh': True},
+            ["dbt", "run", "--full-refresh"]),
+        ("dbt", "run", {'full_refresh': 3}, TypeError()),
+        ("dbt", "run", {'full_refresh': 'hello'}, TypeError()),
+        ("dbt", "run", {'profile': 'test_profile'},
+            ["dbt", "run", "--profile", "test_profile"]),
+
+        # docs specific params
+        ("dbt", "docs", {'no_compile': True},
+            ["dbt", "docs", "--no-compile"]),
+
+        # debug specific params
+        ("dbt", "debug", {'config_dir': '/path/to/config_dir'},
+            ["dbt", "debug", "--config-dir", '/path/to/config_dir']),
+
+        # ls specific params
+        ("dbt", "ls", {'resource_type': '/path/to/config_dir'},
+            ["dbt", "ls", "--resource-type", '/path/to/config_dir']),
+        ("dbt", "ls", {'select': 'my_model'},
+            ["dbt", "ls", "--select", "my_model"]),
+        ("dbt", "ls", {'exclude': 'my_model'},
+            ["dbt", "ls", "--exclude", "my_model"]),
+        ("dbt", "ls", {'output': 'my_model'},
+            ["dbt", "ls", "--output", "my_model"]),
+        ("dbt", "ls", {'output_keys': 'my_model'},
+            ["dbt", "ls", "--output-keys", "my_model"]),
+
+        # rpc specific params
+        ("dbt", "rpc", {'host': 'http://my-host-url.com'},
+            ["dbt", "rpc", "--host", 'http://my-host-url.com']),
+        ("dbt", "rpc", {'port': '8080'}, TypeError()),
+        ("dbt", "rpc", {'port': 8080}, ["dbt", "rpc", "--port", '8080']),
+
+        # run specific params
+        ("dbt", "run", {'fail_fast': True}, ["dbt", "run", "--fail-fast"]),
+
+        # test specific params
+        ("dbt", "test", {'data': True}, ["dbt", "test", '--data']),
+        ("dbt", "test", {'schema': True}, ["dbt", "test", '--schema']),
+        # without params
+
+    ]
+)
+def test_create_cli_command_from_params(
```

Comment:
Cred to you for thorough testing of this function!

Reply:
Almost all of the test descriptions I just added and pretty much all of the test data too where suggested/generated by github copilot.

It's so accurate it get's a little scary